exchanges/persisted-fetch/src/persistedFetchExchange.ts

```diff
@@ -90,24 +99,85 @@ const makePersistedFetchSource = (
         },
       };
 
-      return makeFetchSource(
+      const url = makeFetchURL(operation, { ...body, query: '' });
+      const fetchOptions = makeFetchOptions(operation, body);
+
+      dispatchDebug({
+        type: 'fetchRequest',
+        message: 'A fetch request for a persisted query is being executed.',
         operation,
-        makeFetchURL(operation, { ...body, query: '' }),
-        makeFetchOptions(operation, body)
+        data: {
+          url,
+          fetchOptions,
+        },
+      });
+
+      return pipe(
+        makeFetchSource(operation, url, fetchOptions),
+        onPush(result => {
+          const persistFail =
+            result.error &&
+            (isPersistedMiss(result.error) ||
+              isPersistedUnsupported(result.error));
+          const error = !result.data ? result.error : undefined;
+
+          dispatchDebug({
+            // TODO: Assign a new name to this once @urql/devtools supports it
+            type: persistFail || error ? 'fetchError' : 'fetchSuccess',
+            message: persistFail
+              ? 'A Persisted Query request has failed. A non-persisted GraphQL request will follow.'
+              : `A ${
+                  error ? 'failed' : 'successful'
+                } fetch response has been returned.`,
+            operation,
+            data: {
+              url,
+              fetchOptions,
+              value: persistFail ? result.error! : error || result,
+            },
+          });
+        })
       );
     })
   );
 };
 
 const makeNormalFetchSource = (
-  operation: Operation
+  operation: Operation,
+  dispatchDebug: ExchangeInput['dispatchDebug']
 ): Source<OperationResult> => {
   const body = makeFetchBody(operation);
+  const url = makeFetchURL(operation, body);
+  const fetchOptions = makeFetchOptions(operation, body);
 
-  return makeFetchSource(
+  dispatchDebug({
```

Comment:
The 1-1 mapping you've got here will definitely work - lets catch up on the other stuff this week!